docs/src/guide/periodic_problems.jl

```diff
@@ -0,0 +1,321 @@
+#
+# Let us introduce some jargon: The periodicity of our problem implies that we may define
+# a **lattice**
+# ```
+#           -a       -a/2      +a/2       +a
+#        ... |---------|---------|---------| ...
+#                 a         a         a
+# ```
+# with lattice constant ``a``. Each cell of the lattice is an identical periodic image of
+# any of its neighbours. For finding ``f`` it is thus sufficient to consider only the
+# problem inside the **unit cell** ``[-a/2, a/2)``. In passing we note that the definition
+# of the unit cell is arbitrary up to translations. A choice ``[0, a)``, for example,
+# would have done just as well.
+#
+# ## Periodic operators and the Bloch transform
+# Not only functions, but also operators can feature periodicity.
+# Consider for example the **free-electron Hamiltonian**
+# ```math
+#     H = -\frac12 Δ.
+# ```
+# The free-electron model is a model in which the electron motion is only governed
+# by their own kinetic energy. As this model features no potential which could make one point
+# in space more preferred than another, we would expect this model to be periodic.
+# If an operator is periodic with respect to a lattice like the one defined above,
+# than it commutes with all lattice translations. For the free-electron case ``H``
+# one can easily show exactly that, i.e.
+# ```math
+#    T_{na} H = H T_{na} \quad  ∀ n ∈ \mathbb{Z}.
+# ```
+#
+# **Bloch's theorem** now tells us that for such types of operators,
+# the solutions to the eigenproblem
+# ```math
+#     H v_{kj} = ε_{kj} v_{kj}
+# ```
+# satisfy a factorisation
+# ```math
+#     v_{kj}(x) = e^{i k⋅x} ψ_{kj}(x)
+# ```
+# into a plane wave ``e^{i k⋅x}`` and a lattice-periodic function
+# ```math
+#    T_{na} ψ_{kj}(x) = ψ_{kj}(x - na) = ψ_{kj}(x) \quad ∀ n ∈ \mathbb{Z}.
+# ```
+# In this ``j`` is a labelling integer index and ``k`` is a real number,
+# whose details will be clarified in the next section.
+# Functions ``v_{kj}`` satisfying this factorisation are also known as
+# **Bloch functions** or **Bloch states**.
+#
+# Applying ``2H = -Δ = - \frac{d^2}{d x^2}`` to such a Bloch wave yields
+# ```math
+# \begin{aligned}
+# -\Delta \left(e^{i k⋅x} ψ_{kj}(x)\right)
+#    &= k^2 e^{ik⋅x} ψ_{kj}(x) - 2ik e^{ik⋅x} (∇ψ_{kj}(x)) - e^{ik⋅x} (Δψ_{kj}(x)) \\
+#    &= e^{ik⋅x} (k^2 - 2ik ∇ - Δ) ) ψ_{kj}(x) \\
+#    &= e^{i k⋅x} (-i∇ + k)^2 ψ_{kj}(x) \\
+#    &= e^{i k⋅x} 2H_k ψ_{kj}(x),
+# \end{aligned}
+# ```
+# where we used
+# ```math
+#     H_k = \frac12 (-i∇ + k)^2.
+# ```
+# The action of this operator on a function ``ψ_{kj}`` is given by
+# ```math
+#     H_k ψ_{kj} = e^{-i k⋅x} H e^{i k⋅x} ψ_{kj},
+# ```
+# which in particular implies that
+# ```math
+#    H_k ψ_{kj} = ε_{kj} ψ_{kj} \quad ⇔ \quad H (e^{i k⋅x} ψ_{kj}) = ε_{kj} (e^{i k⋅x} ψ_{kj}).
+# ```
+# To seek the eigenpairs of ``H`` we may thus equivalently
+# find the eigenpairs of *all* ``H_k``.
```

Comment:
Say that the point of this is that the eigenvectors of H_k are periodic and can therefore be represented on a single unit cell (in contrast to those of H)